<launch>

	<arg name="dry_run" default="false" />

  <arg name="use_sim_time" default="false" />

  <node pkg="asb_canopy_spraying_task" exec="canopy_volume_estimation_node" name="canopy_volume_estimation">
    <remap from="points_in" to="/scan_front_multilayer/points_filtered"/>
    <param name="use_sim_time" value="$(var use_sim_time)" />
    <param name="resolution" value="0.05" />
    <param name="hit_count_threshold" value="5" />
    <param name="max_range" value="5.0" />
  </node>

  <node pkg="asb_canopy_spraying_task" exec="spraying_regulator.py" name="spraying_regulator">
    <param name="max_canopy_width" value="2.0" />
    <param name="canopy_layer_bounds" value="[0.2, 0.6, 1.0, 1.4, 1.8, 2.2]" />
  </node>

  <node pkg="asb_canopy_spraying_task" exec="spraying_task_plan_executor.py" name="spraying_task_plan_executor">
    <param name="task_plan_file_path" value="$(find-pkg-share asb_canopy_spraying_task)/config/local_data/cornaredo/task_plan.yaml" />
    <param name="dry_run" value="$(var dry_run)" />
  </node>

</launch>
